charts/victoria-metrics-cluster/templates/vmselect-deployment.yaml

```diff
@@ -76,20 +76,28 @@ spec:
               containerPort: {{ include "split-host-port" .Values.vmselect.extraArgs.clusternativeListenAddr }}
           {{- end }}
           readinessProbe:
+            {{- if not .Values.vmselect.probe.readiness.custom }}
             httpGet:
             {{- if index  .Values.vmselect.extraArgs "http.pathPrefix" }}
               path: {{ trimSuffix "/" (index .Values.vmselect.extraArgs "http.pathPrefix") }}/health
             {{- else }}
               path: /health
             {{- end }}
               port: http
+            {{- else }}
+              {{- toYaml .Values.vmselect.probe.readiness.custom | nindent 12 }}
+            {{- end }}
```

Comment:
Can we move values for probe into values and just use `{{ .Values.vmselect.probe.readiness }}` then?
Combining this with [`tpl` filter](https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function) should allow to keep logic with automatic handling of `http.pathPrefix` parameter.

This allows overriding any parameters to customize probe and avoid if/else nesting.

WDYT @Haleygo @jifwin ?

Reply:
Is that what you mean?
values.yaml
`
  probe:
    readiness: |
      httpGet:
        {{- if index  .Values.vminsert.extraArgs "http.pathPrefix" }}
        path: {{ trimSuffix "/" (index .Values.vminsert.extraArgs "http.pathPrefix") }}/health
        {{- else }}
        path: /health
        {{- end }}
        port: http
      initialDelaySeconds: 5
      periodSeconds: 15
      timeoutSeconds: 5
      failureThreshold: 3
`

vminsert-deployment.yaml
`
          readinessProbe: {{ tpl .Values.vminsert.probe.readiness . | nindent 12}}
`

Reply:
Yes, exactly. 
I'd like to ask @Haleygo for the feedback regarding which option seems to be suite this case better.

Reply:
It's worth nothing that it would break existing overrides of vm*.probe.readiness, as it would no longer be an object, but a string instead.

Reply:
You're right, somehow I've missed this completely, in this case just ignore this suggestion :)